components/ai_chat/core/browser/model_validator.cc

```diff
@@ -5,14 +5,73 @@
 
 #include "brave/components/ai_chat/core/browser/model_validator.h"
 
+#include <string>
+
 #include "base/numerics/safe_math.h"
+#include "brave/components/ai_chat/core/common/features.h"
 #include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom.h"
 #include "brave/net/base/url_util.h"
+#include "net/base/ip_address.h"
 
 class GURL;
 
 namespace ai_chat {
 
+namespace {
+bool IsValidPrivateHost(const GURL& endpoint) {
+  std::string host = endpoint.host();
+
+  // Ensure the hostname is valid (non-empty and doesn't start with a dot)
+  if (host.empty() || host.front() == '.') {
+    return false;
+  }
+
+  std::string lower_host = base::ToLowerASCII(host);
+
+  // Check if the host ends with ".local"
+  if (base::EndsWith(lower_host, ".local", base::CompareCase::SENSITIVE)) {
```

Comment:
`.local` addresses might resolve to external hosts. If it's `.local` we should first resolve, and then replace the `.local` address to the resolved local address, at that particular point in time

Reply:
Ad-hoc resolution is a bit out of my comfort zone right now, following a cursory review of `net::HostResolver`. By default, Leo supports external hosts, so should we be too concerned if a `*.local` address resolves to an external IP address? If this is an important issue, the easiest path right now might be to simply defer support for `*.local` addresses to a later date, or trust that the user has verified their DNS configuration. Worth noting that all of this is behind the experimental flag, so no out-of-the-box support for `*.local` domains is being introduced either way.

Reply:
I'd rather defer `*.local` resolutions to a later stage, indeed. The point is that the behavior on `.local` addresses is not stable, and depends on how it's resolved. macOS does indeed have a stable-ish resolution process; it's not the case for Microsoft or Linux-based OSes.

I'd rather dig a bit more in the Chromium source code, to see if somebody already tried to generalize this link-local resolution across OSes.

Reply:
I've [removed](https://github.com/brave/brave-core/pull/26475/commits/f29cbd4ecc42f4b423e2564f3ba24cbbf753ed97) support for `.local` domains for now, and will track the effort moving forward with [#42367](https://github.com/brave/brave-browser/issues/42367).